### PR TITLE
Refactor cloud-agent module logs for Log4j2

### PR DESCRIPTION
**Pull request by KlausDornsbach**
### Description

With the new version of Log4j there is some space for improving log readability using the new features. This PR refactors the logs from the cloud-agent module.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [X] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [X] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### How Has This Been Tested?
It was tested in a local lab:
- I have done some operations on cloudstack to induce the generation of the logs
- No problems or changes to the log output were found

### CODE REVIEW

File: agent/src/main/java/com/cloud/agent/Agent.java [hunk 1]
```diff
@@ -286,22 +287,22 @@ public void start() {
         try {
             _connection.start();
         } catch (final NioConnectionException e) {
-            logger.warn("NIO Connection Exception  " + e);
+            logger.warn("NIO Connection Exception {}", e);
```
Comment (DaanHoogland) [inline agent/src/main/java/com/cloud/agent/Agent.java:290]: a `warn()` with an exception and then an `info()` with a more generic message. I think we can improve more here.
i.e:
- warn with `e.getLocalizedMessage()` and
- remove the `info()` which is of lesser level anyway.

**Comment by DaanHoogland** [comment]
@blueorangutan package

File: agent/src/main/java/com/cloud/agent/Agent.java [hunk 2]
```diff
@@ -312,7 +313,7 @@ public void start() {
     }
 
     public void stop(final String reason, final String detail) {
-        logger.info("Stopping the agent: Reason = " + reason + (detail != null ? ": Detail = " + detail : ""));
+        logger.info("Stopping the agent: Reason = {}", reason + ObjectUtils.defaultIfNull(": Detail = " + detail, ""));
```
Comment (JoaoJandre) [inline agent/src/main/java/com/cloud/agent/Agent.java:316]: This defaultIfNull does not make sense, `": Detail = " + detail` will never be null. Something like this makes more sense, but maybe using string format would be better for the second parameter.
```suggestion
        logger.info("Stopping the agent: Reason = {} {}", reason, () -> ": Detail = "  + ObjectUtils.defaultIfNull(detail, ""));
```

File: agent/src/main/java/com/cloud/agent/Agent.java [hunk 3]
```diff
@@ -396,17 +397,16 @@ private synchronized void scheduleHostLBCheckerTask(final long checkInterval) {
...
             hostLBTimer.scheduleAtFixedRate(new PreferredHostCheckerTask(), checkInterval, checkInterval);
         }
     }
 
     public void scheduleWatch(final Link link, final Request request, final long delay, final long period) {
         synchronized (_watchList) {
-            if (logger.isDebugEnabled()) {
-                logger.debug("Adding a watch list");
-            }
+            logger.debug("Adding a watch list");
```
Comment (JoaoJandre) [inline agent/src/main/java/com/cloud/agent/Agent.java:408]: We could rewrite this log to add the task that is being added on the watchList.

File: agent/src/main/java/com/cloud/agent/Agent.java [hunk 4]
```diff
@@ -377,7 +378,7 @@ public Long getId() {
     }
 
     public void setId(final Long id) {
-        logger.info("Set agent id " + id);
+        logger.info("Set agent id {}", id);
```
Comment (JoaoJandre) [inline agent/src/main/java/com/cloud/agent/Agent.java:381]:
```suggestion
        logger.debug("Set agent id {}", id);
```

File: agent/src/main/java/com/cloud/agent/Agent.java [hunk 5]
```diff
@@ -432,9 +432,7 @@ protected void cancelTasks() {
             for (final WatchTask task : _watchList) {
                 task.cancel();
             }
-            if (logger.isDebugEnabled()) {
-                logger.debug("Clearing watch list: " + _watchList.size());
-            }
+            logger.debug("Clearing watch list: {}", _watchList.size());
```
Comment (JoaoJandre) [inline agent/src/main/java/com/cloud/agent/Agent.java:435]: Either log the entire list, or change the log to specify that the size is being logged

**Review by JoaoJandre — COMMENTED** [review]
Overall looks good, I've found one issue that should be fixed and some points that could be improved but are a little out of scope of this PR.

File: agent/src/main/java/com/cloud/agent/Agent.java (hunk 3 above)
Comment (KlausDornsbach) [reply agent/src/main/java/com/cloud/agent/Agent.java:408]: logger.debug("Adding task with request: {} to watch list", request.toString());
something like this?

**Review by DaanHoogland — APPROVED** [review]
clgtm

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Review by JoaoJandre — APPROVED** [review]
CLGTM

**Comment by JoaoJandre** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@KlausDornsbach 
```
13:59:26 [ERROR] /jenkins/workspace/acs-centos7-pkg-builder/dist/rpmbuild/BUILD/cloudstack-4.20.0.0-SNAPSHOT/server/src/main/java/com/cloud/storage/snapshot/SnapshotSchedulerImpl.java:[420,12] error: cannot find symbol
13:59:26   symbol:   variable s_logger
13:59:26   location: class SnapshotSchedulerImpl
```
I can not find the reference to `s_logger` in `SnapshotSchedulerImpl` in main, so maybe rebase on main would solve the issue.

**Comment by KlausDornsbach** [comment]
Thanks for the heads up @DaanHoogland, hopefully issues are solved now.

**Comment by JoaoJandre** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test alma9 kvm-alma9

**Comment by JoaoJandre** [comment]
Merging based on approvals and CI result

**Pull request merged**